src/Core/Settings.cpp

```diff
@@ -360,6 +360,8 @@ namespace ErrorCodes
     M(Int64, distributed_ddl_task_timeout, 180, "Timeout for DDL query responses from all hosts in cluster. If a ddl request has not been performed on all hosts, a response will contain a timeout error and a request will be executed in an async mode. Negative value means infinite. Zero means async mode.", 0) \
     M(Milliseconds, stream_flush_interval_ms, 7500, "Timeout for flushing data from streaming storages.", 0) \
     M(Milliseconds, stream_poll_timeout_ms, 500, "Timeout for polling data from/to streaming storages.", 0) \
+    M(UInt64, min_free_disk_bytes_to_throw_insert, 0, "Minimum free disk space bytes to throw an insert.", 0) \
```

Comment:
Yes, I have been using this name because I was guiding myself by what was originally described in the issue #65951 but having the 'throw' word (even in the setting) can be misleading as it may cause some confusion on wether the insert is performed or an exception is thrown. Will change the setting name as well